### PR TITLE
detect (and short circuit) file limitations at file scope

### LINKED ISSUE #583: match "pure" file scope rules first to enable fast limitation detection
in #390 we propose to use capa rules to identify files that can't really be analyzed fully (e.g. .NET modules or packed programs). for these samples, we probably don't want to do a full analysis, because it may take a while (esp with packed programs) and the results will be inconclusive/inaccurate.

most of these samples can be identified with "pure" file features. that is, file-scope features that are not matches to more specific scopes. e.g. `string`, `section`, etc.  notably, these features can typically be extracted quickly without a full analysis pass.

capa could identify "pure" file scope rules and run these first against each sample. then, if there's a limitation identified, capa can short circuit and exit without doing an analysis pass. if no limitations are identified, the "impure" file scope rules can be run as they do today - after the insn/bb/function scope rules.

pro:
  - capa can run more quickly, especially against difficult, packed samples that will yield few/no results anyways

con:
  - more logic needed, to identify "pure" rules and invoke them at the right time

while i think `capa.main` would become more complicated, this behavior is "opt-in" and capa-as-a-library users wouldn't not necessarily have to do anything differently, unless they wanted to also short circuit early.

### PR DESCRIPTION
**Pull request by williballenthin**
Do an initial pass at only the file scope using a new "pefile" feature extractor. This is quick work. If there are any file limitations detected at this point, avoid doing a full code analysis, because we probably won't get very far. This will enable use to better handle/avoid vmprotect samples, such as in #509 

closes #583 

@evandrix

### CODE REVIEW

File: tests/test_smda_features.py [hunk 1]
```diff
@@ -14,39 +14,25 @@
...
-            ret.append(pytest.param(*values, marks=pytest.mark.xfail(reason="SMDA bug tracked #585", strict=True)))
-        elif sample == "al-khaser x64" and sys.platform == "win32":
-            ret.append(pytest.param(*values, marks=pytest.mark.xfail(reason="SMDA bug tracked #585", strict=True)))
-        else:
-            ret.append(values)
-    return parametrize(params, ret, **kwargs)
-
-
-@smda_parametrize(
+@parametrize(
```
Comment (williballenthin) [inline tests/test_smda_features.py:17]: semi-unrelated fix as i figured out how to do this better in the pefile work.

**Review by mr-tz — APPROVED** [review]
very cool!

**Review by Ana06 — APPROVED** [review]
👏  (CHANGELOG update missing)
